components/ILIAS/Certificate/classes/class.ilObjCertificateSettings.php

```diff
@@ -30,223 +33,81 @@
-
-                $this->certificate_repo->updateDefaultBackgroundImagePaths($old_relative_path, $new_relative_path);
-
-                if (
-                    !$this->certificate_repo->isBackgroundImageUsed($old_relative_path)
-                    && !$this->user_certificate_repo->isBackgroundImageUsed($old_relative_path)
-                ) {
-                    if (is_file($old_path) && file_exists($old_path)) {
-                        unlink($old_path);
-                    }
-
-                    if (is_file($old_path_thumb) && file_exists($old_path_thumb)) {
-                        unlink($old_path_thumb);
-                    }
-                }
-                return true;
-            }
-
-            $this->cert_logger->error(sprintf(
-                "Final background image '%s' does not exist or is empty",
-                $new_image_path
-            ));
+        if ($rid = $this->irss->manage()->find($id)) {
+            return $rid;
         }
-
-        return false;
-    }
-
-    public function deleteBackgroundImage(): bool
-    {
-        $result = true;
-
-        if (
-            $this->certificate_settings->get('defaultImageFileName', '')
-            && !$this->certificate_repo->isBackgroundImageUsed($this->getDefaultBackgroundImagePath(true))
-            && !$this->user_certificate_repo->isBackgroundImageUsed($this->getDefaultBackgroundImagePath(true))
-        ) {
-            //No certificates exist using the currently configured file, deleting file possible.
-
-            if (is_file($this->getDefaultBackgroundImageThumbPath())) {
-                $result &= unlink($this->getDefaultBackgroundImageThumbPath());
-            }
-            if (is_file($this->getDefaultBackgroundImagePath())) {
-                $result &= unlink($this->getDefaultBackgroundImagePath());
-            }
-
-            foreach (ilCertificateBackgroundImageFileService::VALID_BACKGROUND_IMAGE_EXTENSIONS as $extension) {
-                if (is_file($this->getDefaultBackgroundImageTempFilePath($extension))) {
-                    $result &= unlink($this->getDefaultBackgroundImageTempFilePath($extension));
-                }
-            }
+        if ($id !== '') {
+            $id = $this->getBackgroundImageDefaultFolder() . $id;
         }
-
-        $this->certificate_settings->set('defaultImageFileName', '');
-
-        /** @noinspection PhpCastIsUnnecessaryInspection */
-        /** @noinspection UnnecessaryCastingInspection */
-        return (bool) $result; // Don't remove the cast, otherwise $result will be 1 or 0
-    }
-
-    public function getBackgroundImageDefaultFolder(bool $relativePath = false): string
-    {
-        return ($relativePath ? '' : CLIENT_WEB_DIR) . '/certificates/default/';
+        if ($this->filesystem->has($id)) {
+            return ilWACSignedPath::signFile(ILIAS_HTTP_PATH . '/public/' . ILIAS_WEB_DIR . '/' . CLIENT_ID . $id);
```

Comment:
`public` will IMO not be part of the HTTP path, if ILIAS is configured correctly.